stylus-proc/src/macros/public/export_abi.rs

```diff
@@ -0,0 +1,164 @@
+// Copyright 2024, Offchain Labs, Inc.
+// For licensing, see https://github.com/OffchainLabs/stylus-sdk-rs/blob/main/licenses/COPYRIGHT.md
+
+use proc_macro2::TokenStream;
+use quote::quote;
+use syn::parse_quote;
+
+use crate::types::Purity;
+
+use super::types::{FnArgExtension, FnExtension, InterfaceExtension, PublicImpl};
+
+#[derive(Debug)]
+pub struct InterfaceAbi;
+
+impl InterfaceExtension for InterfaceAbi {
+    type FnExt = FnAbi;
+    type Ast = syn::ItemImpl;
+
+    fn build(_node: &syn::ItemImpl) -> Self {
+        InterfaceAbi
+    }
+
+    fn codegen(iface: &PublicImpl<Self>) -> Self::Ast {
+        let PublicImpl {
+            generic_params,
+            self_ty,
+            where_clause,
+            inheritance,
+            funcs,
+            ..
+        } = iface;
+
+        let name = match self_ty {
+            syn::Type::Path(path) => path.path.segments.last().unwrap().ident.clone().to_string(),
+            _ => todo!(),
```

Comment:
Why todo? Will this change later on? If so, maybe track with with a code comment or linear task, such as TODO(LINEAR_ITEM_NUMBER)